### ISSUE TITLE
Feature Request: task dependencies on other tasks

### ISSUE THREAD

**Issue by freakinhippie** [fabiospampinato/vscode-todo-plus#272]
It would be very helpful if tasks could reference other tasks as dependencies. Simple dependencies within a project can already be achieved with task ordering, but when there are dependencies on other projects there is no obvious way to track them. I'm not certain how it could be implemented since the tasks don't have identifiers though.

**Comment by fabiospampinato** [comment]
> I'm not certain how it could be implemented since the tasks don't have identifiers though.

Yeah. Maybe potentially we could assign a incrementing counter to each task and then use that to reference them, kind of like in github.

**Comment by freakinhippie** [comment]
That sounds reasonable to me, though I'm not sure what impact it would have on other people's workflow. Thanks for considering it.

**Comment by sahil48** [comment]
That could work. Would the id's be in a separate text file, outside of the document, or in the header?

What about an id `@id` tag assigned to the original task and reference tag `@ref` to the task that will have the reference when the user chooses to reference the task, possibly with some short user-defined name for the reference and an automatic number? When assigning the reference, the user types to search for the task and then selects it. I'm assuming TODO has a searchable index of all tasks somewhere since there is a sidebar for searching tasks already.

**Comment by freakinhippie** [comment]
I like the `@id` and `@ref` suggestions because I think the information should be kept with the tasks themselves.

**Comment by fabiospampinato** [comment]
> That sounds reasonable to me, though I'm not sure what impact it would have on other people's workflow. Thanks for considering it.

It would be opt-in.

> That could work. Would the id's be in a separate text file, outside of the document, or in the header?

I think the best way would be to put something like this: `#number` at the end of each todo.

This is pretty big change though, I don't think I will have time to implement it, possibly ever, somebody else would need to do the work on this for this to happen.
